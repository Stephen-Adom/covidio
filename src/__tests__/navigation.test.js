import { render, screen } from '@testing-library/react';
import { RouterProvider } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavBar from '../features/Navigation/navigation';
import router from '../route/routes';

describe('Navigation', () => {
  it('should render navbar correctly', () => {
    render(
      <RouterProvider router={router}>
        <NavBar />
      </RouterProvider>,
    );

    const headerElement = screen.getByRole('heading');

    expect(headerElement).toBeInTheDocument();
  });

  it('should render navbar list items correctly', async () => {
    const tree = renderer.create(
      <RouterProvider router={router}>
        <NavBar />
      </RouterProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
