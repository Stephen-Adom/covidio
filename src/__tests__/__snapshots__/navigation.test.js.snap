// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render navbar list items correctly 1`] = `
<div
  className="App"
>
  <section
    className="splash-container"
  >
    <div
      className="relative logo-container"
    >
      <img
        alt="covido"
        className="bounce-in-fwd"
        src="covido-logo.png"
        width="100%"
      />
      <h1
        className="logo tracking-in-expand"
      >
        Covidio
      </h1>
    </div>
  </section>
</div>
`;
